bundle node refactored to relay on createNode() being creating links

scf2k opened the pull request.

afelix committed:
'hold' -> 'holds'

Commit:
ie6-ie9 techs

Commit:
ieN better variant

Commit:
tab fix

veged committed:
close #79

1. Special case for subtract item with key `=== ''`.
2. In `forEach` skip deps that missed in `items`.
3. Add some test-case.

Commit:
0.4.0

Commit:
- bundle node refactored to relay on createNode() being creating links

arikon commented on lib/nodes/bundle.js:
```diff
@@ -42,18 +42,10 @@ var BundleNode = exports.BundleNode = INHERIT(MagicNode, {
-                    arch.link(n.getId(), optimizer.getId());
-                    arch.link(optimizer.getId(), bundleNode);
-                }
+                console.log('thisNode %s bundleNode %s', thisNode, bundleNode);
```
`console.log()`

arikon commented on lib/nodes/bundle.js:
```diff
@@ -68,19 +60,35 @@ var BundleNode = exports.BundleNode = INHERIT(MagicNode, {
+            arch = ctx.arch;
+
+        if (this['create-' + tech + '-node']) node = this['create-' + tech + '-node'](ctx, tech)
+        else node = this.createDefaultNode(ctx, tech);
```
`node = (this['create-' + tech + '-node'] || this.createDefaultNode)(ctx, tech);`

arikon commented on lib/nodes/bundle.js:
```diff
@@ -68,19 +60,35 @@ var BundleNode = exports.BundleNode = INHERIT(MagicNode, {
+            arch = ctx.arch;
+
+        if (this['create-' + tech + '-optimizer-node']) node = this['create-' + tech + '-optimizer-node'](ctx, sourceNode)
+        else node = this.createDefaultOptimizerNode(ctx, sourceNode, tech);
```
`node = (this['create-' + tech + '-optimizer-node'] || this.createDefaultOptimizerNode)(ctx, sourceNode, tech);`

arikon committed:
Rename Registry to NodesRegistry

arikon committed:
Fix crash with empty .bem/make.js

arikon committed:
More precise logging in lib/server.js

Commit:
- library node is considered valid for 1 minute since last execution

arikon committed:
Actualize dependencies on apw and node

arikon committed:
0.5.0

Commit:
- removed testing code

arikon committed:
0.5.1

Commit:
- pass node id not node object to unlink
- check for cleanup method

Commit:
- winston is added as logging engine
- verbose argument for make and server commands accepts one of the predefined logging levels
- console.log is replaced with winston call where appropriate

Commit:
- fixed issue with logging levels
- added helpers to trace methods execution

Commit:
- removed barbage code

Commit:
- jsdoc updated
- internal filter added (to avoid wrapping of the properties like __self)
- the white and black lists are arrays for now

Commit:
- jsdoc updated
- verbose param renamed to verbosity

Commit:
- added default verbosity level to help output

Commit:
- use f* methods instead of normal ones with nested util.format()

Commit:
- use double quotes

Commit:
- more refactoring

Commit:
Merge branch 'server' of github.com:bem/bem-tools into bundle-refactoring

Conflicts:
	lib/nodes/bundle.js

Commit:
- rollback

Commit:
- spacing fixed

arikon commented on lib/nodes/bundle.js:
```diff
@@ -132,68 +119,125 @@ var BundleNode = exports.BundleNode = INHERIT(MagicNode, {
+        magicNode && arch.link(magicNode, buildNodeId);
+
+        if (!arch.hasNode(metaNode.getId())) {
+            arch.setNode(metaNode, buildNodeId, this.createNode(ctx, declTech));
```
Сюда надо перенести коммент

`// FIXME: rethink of this.createNode(ctx, declTech).getId() !!!`

Commit:
- _*.ie.css is dependant on *.css for now (close #99)
